packages/library-base/src/commonMain/kotlin/io/realm/internal/BaseRealmImpl.kt

```diff
@@ -114,9 +112,9 @@ abstract class BaseRealmImpl internal constructor(
 
     // Not all sub classes of `BaseRealm` can be closed by users.
     internal open fun close() {
-        val reference = realmReference
-        reference.checkClosed()
-        RealmInterop.realm_close(reference.dbPointer)
-        log.info("Realm closed: ${configuration.path}")
+        realmReference.close()
```

Comment:
Had some issue getting all references closed without being able to close some of them multiple times. After bookkeeping the references in the notifier/writer we are now back at closing reference once and enforcing this by calling `checkClose` in `RealmReference.close()`. Only minor tricky thing is that the user facing realm is now not responsible for closing its own `realmReference` as this is actually one of the references otherwise managed by the user facing realms version tracker, the writer or the notifier.